shortcuts-raycast-extension/src/view/shortcuts-list.tsx

```diff
@@ -0,0 +1,90 @@
+import { Action, ActionPanel, closeMainWindow, getPreferenceValues, Icon, List, PopToRootType } from "@raycast/api";
+import { KeymapDropdown } from "./keymap-dropdown";
+import { generateHotkeyText } from "./hotkey-text-formatter";
+import { Application, Keymap, Section, SectionShortcut } from "../model/internal/internal-models";
+import { runShortcuts } from "../engine/shortcut-runner";
+import useKeyCodes from "../load/key-codes-provider";
+import { useEffect, useState } from "react";
+
+interface ShortcutsListProps {
+  application: Application | undefined;
+}
+
+interface Preferences {
+  delay: string;
+}
+
+export function ShortcutsList({ application }: ShortcutsListProps) {
+  const keyCodesResponse = useKeyCodes();
+  const [keymaps, setKeymaps] = useState<string[]>([]);
```

Comment:
I think no state is required here, since keymaps is never altered from within the component, and it can be retrieved synchronously. See a minor refactor here (last 2 commits) https://github.com/th0rgall/hotkys/commits/web-app-shortcuts-review/

Reply:
agree